### PR TITLE
Compression codecs refactoring

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Continuation of #24424.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
-5 lines of code!

**Comment by alexey-milovidov** [comment]
> https://clickhouse-test-reports.s3.yandex.net/25002/8d851457254cf7fe90edde613cffeeeb011495eb/performance_comparison/report.html#fail1

The difference in `sumMap` performance is Ok.

**Comment by alexey-milovidov** [comment]
@mergifyio update

File: tests/integration/test_non_default_compression/configs/allow_experimental_codecs.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<yandex>
+    <profiles>
+        <default>
+            <allow_experimental_codecs>1</allow_experimental_codecs>
```
Comment (filimonov) [inline tests/integration/test_non_default_compression/configs/allow_experimental_codecs.xml:4]: Does nothing now, but ok. Lets keep place for experiments :)

**Pull request merged**